torchgeo/losses/ssl.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Loss functions self-supervised learning."""
+
+import torch
+import torch.nn.functional as F
+from torch.nn.modules import Module
+
+
+class NTXentLoss(Module):
```

Comment:
Should we name this NT-Xent or InfoNCE? I see both very commonly used (SimCLR calls it NT-Xent, MoCo calls it InfoNCE). I believe MoCo v1 links to a specific paper that "defines" this loss. We should add a citation for that. Regardless of which we call it, we should add an "aka" in the docstring.

Reply:
NT-Xent is different than InfoNCE. NT-Xent normalizes using a temperature value and also assumes the target is an identity matrix (pairs from the same sample are positives, all else are negatives).

Reply:
Gotcha. Can you implement both?

Reply:
Yes but I think we should do infoNCE in a separate PR. It will take some discussion because it requires that you know where all the positive pairs are in the batch which can be N-pairs.